### ISSUE TITLE
Color picker doesn't work in custom settings

### ISSUE THREAD

**Issue by tandrieuxx** [OCA/web#508]
I haven't managed to make the color picker work on a custom settings page for my own module. All it does is changing the background color of the text input field, but not dynamically, I need to refresh the page to see the new color. When I click on it, the color picker doesn't show up but I can see the page height growing by 1px. I tried to edit manually an existing settings page (Settings -> General Settings) by adding just "widget='color'" to an existing field, and it works, I can see the color picker and use it normally.
Here are some extracts of my code : 
```
class MyModel(models.Model):
    _name = "mymodule.mymodel"

    color = fields.Char()
    image = fields.Binary(attachment=True)

class Settings(models.TransientModel):
    _inherit = "res.config.settings"
    _name = "mymodule.settings"

    color = fields.Char(string="Color")
    image = fields.Binary(string="Image")

    default_settings = {
        'color': "#337AB7",
    }

    @api.model
    def get_default_values(self, fields):
        if self.env["mymodule.mymodel"].search_count([]) > 0:
            m = self.env["mymodule.mymodel"].search([])[0]

            return {
                'color': m.color,
                'image': m.image,
            }
        else:
            return self.default_settings

    @api.model
    def set_values(self):
        if self.env["mymodule.mymodel"].search_count([]) == 0:
            self.env["mymodule.mymodel"].create({
                "color": self.color,
                "image": self.image,
            })
        else:
            m = self.env["mymodule.mymodel"].search([])[0]
            m.color = self.color
            m.image= self.image`
```

```
<record id="mymodule.settings_view_form" model="ir.ui.view">
            <field name="name">mymodule.settings.form</field>
            <field name="model">mymodule.settings</field>
            <field name="arch" type="xml">
                <form string="myform" class="oe_form_configuration">
                    <header>
                        <button string="Valider" type="object"
                            name="execute" class="oe_highlight"/>
                        <button string="Annuler" type="object"
                            name="cancel" class="oe_link"/>
                    </header>
                    <group string="Couleurs">
                        <label for="color"/>
                        <div>
                            <field name="color" widget="color" class="oe_inline"/>
                        </div>
                    </group>
                    <group string="Image">
                        <div>
                            <field name="image" widget="image" class="oe_left" options="{'size': [400, 225]}"/>
                        </div>
                    </group>
                </form>
            </field>
        </record>
```

**Comment by pedrobaeza** [comment]
If it's working on other field, it's working correctly, so there's not a bug. Follow README instructions and you will get it, or learn what is the problem in your code.

**Comment by tandrieuxx** [comment]
I've tried to put the same XML code on a basic form view, it works. It doesn't if it's a configuration form. I declared my char field exactly like README says (even by adding size=7), and my field with a widget="color" in XML. There's not much more to do actually. The HTML code I get in my browser is exactly the same (near text input field) as I got with a regular form view (where widget worked), except I can't see any change when I click on the field. The background color is changing when I refresh the page though. So why is only half the widget working ? The only difference is that it's about settings, so a transient model linked to a regular model.
